rust/noosphere-storage/src/exportable.rs

```diff
@@ -0,0 +1,176 @@
+use crate::{Storage, Store, SPHERE_DB_STORE_NAMES};
+use anyhow::Result;
+use async_trait::async_trait;
+use noosphere_common::ConditionalSync;
+use std::pin::Pin;
+use tokio_stream::{Stream, StreamExt};
+
+#[cfg(target_arch = "wasm32")]
+pub type IterableStream<'a> = Pin<Box<dyn Stream<Item = Result<(Vec<u8>, Option<Vec<u8>>)>> + 'a>>;
+#[cfg(not(target_arch = "wasm32"))]
+pub type IterableStream<'a> =
+    Pin<Box<dyn Stream<Item = Result<(Vec<u8>, Option<Vec<u8>>)>> + Send + 'a>>;
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+pub trait Iterable: Store {
```

Comment:
`Iterable` is an overloaded name -- while we could take the approach of `std`, where generic-named features are referenced via module (`io::Error`, `collections::hash_map::Iter`), maybe here we can append `Store` and `Storage` to most of our storage-specific traits e.g. `Space`.

TODO: Change to `IterableStore`